src/MerkleTree.ts

```diff
@@ -883,7 +908,7 @@ export class MerkleTree extends Base {
 
     let ids : number[]
     if (leaves.every(Number.isInteger)) {
-      ids = leaves.sort((a, b) => a === b ? 0 : a > b ? 1 : -1) // Indices where passed
+      ids = [...leaves].sort((a, b) => a === b ? 0 : a > b ? 1 : -1) // Indices where passed
```

Comment:
is that really needed ?

Reply:
Yes, `.sort` is in place so it was modifying the argument.